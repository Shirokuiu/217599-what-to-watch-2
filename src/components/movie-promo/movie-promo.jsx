import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import MovieControls from "../movie-controls/movie-controls";
import {Header} from "../header/header";

import {Operation} from "../../reducer";

import {withMovieControls} from "../../hocs/with-movie-controls/with-movie-controls";

const MovieControlsWrapped = withMovieControls(MovieControls);

export const MoviePromo = (props) => {
  const PARENT_STATUS = `PROMO`;

  const {promo, history, onUpdateIsFavorite} = props;
  const {backgroundImage, name, posterImage, genre, released, isFavorite, id} = promo;

  const handleFavoriteClick = () => {
    onUpdateIsFavorite();
  };

  return <section className="movie-card">
    <div className="movie-card__bg">
      <img src={backgroundImage} alt={name} />
    </div>

    <h1 className="visually-hidden">WTW</h1>

    <Header />

    <div className="movie-card__wrap">
      <div className="movie-card__info">
        <div className="movie-card__poster">
          <img src={posterImage} alt={name} width="218" height="327"/>
        </div>

        <div className="movie-card__desc">
          <h2 className="movie-card__title">{name}</h2>
          <p className="movie-card__meta">
            <span className="movie-card__genre">{genre}</span>
            <span className="movie-card__year">{released}</span>
          </p>

          {id ? <MovieControlsWrapped
            history={history}
            parent={PARENT_STATUS}
            favorite={isFavorite}
            movieId={id - 1}
            onFavoriteClick={handleFavoriteClick}
          /> : null}
        </div>
      </div>
    </div>
  </section>;
};

MoviePromo.propTypes = {
  promo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    posterImage: PropTypes.string,
    previewImage: PropTypes.string,
    backgroundImage: PropTypes.string,
    backgroundColor: PropTypes.string,
    videoLink: PropTypes.string,
    previewVideoLink: PropTypes.string,
    description: PropTypes.string,
    rating: PropTypes.number,
    scoresCount: PropTypes.number,
    director: PropTypes.string,
    starring: PropTypes.array.string,
    runTime: PropTypes.number,
    genre: PropTypes.string,
    released: PropTypes.number,
    isFavorite: PropTypes.bool,
  }).isRequired,
  history: PropTypes.object.isRequired,
  onUpdateIsFavorite: PropTypes.func
};

const mapDispatchToProps = (dispatch) => ({
  onUpdateIsFavorite: () => {
    dispatch(Operation.loadPromo());
    dispatch(Operation.loadMovies());
  }
});

export default connect(null, mapDispatchToProps)(MoviePromo);

