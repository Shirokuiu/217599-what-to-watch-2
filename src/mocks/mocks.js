export const films = [
  {
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts`
  },
  {
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`
  },
  {
    img: `img/macbeth.jpg`,
    title: `Macbeth`
  },
  {
    img: `img/aviator.jpg`,
    title: `Aviator`
  },
  {
    img: `img/we-need-to-talk-about-kevin.jpg`,
    title: `We need to talk about Kevin`
  }
];
