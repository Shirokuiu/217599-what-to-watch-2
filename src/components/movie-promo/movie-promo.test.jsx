import React from "react";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

import {MoviePromo} from "../movie-promo/movie-promo";

import {films} from "../../mocks/mocks";

Enzyme.configure({adapter: new Adapter()});

it(`Is movie promo rendered`, () => {
  const tree = shallow(<MoviePromo
    promo={films[0]}
    history={{}}
    isAuthorized={false}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
