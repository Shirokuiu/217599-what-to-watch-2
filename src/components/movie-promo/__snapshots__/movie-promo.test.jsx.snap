// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie promo rendered 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="img/the-grand-budapest-hotel-bg.jpg"
    />
  </div>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <Header />
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Comedy
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <Connect(WithMovieControls)
          favorite={false}
          history={Object {}}
          movieId={0}
          onFavoriteClick={[Function]}
          parent="PROMO"
        />
      </div>
    </div>
  </div>
</section>
`;
